src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/ActivitiesService.kt

```diff
@@ -24,34 +29,52 @@ class ActivitiesService(
   private val agencyInternalLocationRepository: AgencyInternalLocationRepository,
   private val programServiceRepository: ProgramServiceRepository,
   private val availablePrisonIepLevelRepository: AvailablePrisonIepLevelRepository,
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val offenderProgramProfileRepository: OffenderProgramProfileRepository,
   private val telemetryClient: TelemetryClient,
 ) {
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun createActivity(dto: CreateActivityRequest): CreateActivityResponse {
+  fun createActivity(dto: CreateActivityRequest): CreateActivityResponse =
 
-    val courseActivity = activityRepository.save(mapModel(dto))
+    activityRepository.save(mapActivityModel(dto)).run {
 
-    val payRates = mapRates(dto, courseActivity)
+      payRates = mapRates(dto, this)
 
-    courseActivity.payRates = payRates
+      telemetryClient.trackEvent(
+        "activity-created",
+        mapOf(
+          "id" to courseActivityId.toString(),
+          "prisonId" to prison.id,
+        ),
+        null
+      )
+      log.debug("Activity created with Nomis id = $courseActivityId")
 
-    telemetryClient.trackEvent(
-      "activity-created",
-      mapOf(
-        "id" to courseActivity.courseActivityId.toString(),
-        "prisonId" to courseActivity.prison.id,
-      ),
-      null
-    )
-    log.debug("Activity created with Nomis id = (${courseActivity.courseActivityId})")
+      CreateActivityResponse(courseActivityId)
+    }
 
-    return CreateActivityResponse(courseActivity.courseActivityId)
-  }
+  fun createOffenderProgramProfile(dto: CreateOffenderProgramProfileRequest): CreateOffenderProgramProfileResponse =
 
-  private fun mapModel(dto: CreateActivityRequest): CourseActivity {
+    offenderProgramProfileRepository.save(mapOffenderProgramProfileModel(dto)).run {
+
+      telemetryClient.trackEvent(
+        "offender-program-profile-created",
+        mapOf(
+          "id" to offenderProgramReferenceId.toString(),
+          "courseActivityId" to courseActivity?.courseActivityId.toString(),
+          "bookingId" to offenderBooking.bookingId.toString(),
+        ),
+        null
+      )
+      log.debug("OffenderProgramProfile created with Nomis id = $offenderProgramReferenceId")
```

Comment:
you probably don't need the debug if you have the telemetry

Reply:
👍